docs/sidechain/spec.md

```diff
@@ -0,0 +1,945 @@
+   a "cross-chain claim" transaction must be use. Note, if the signers list on
+   the door account changes while the signatures are collected, the signers list
+   at the time the quorum is reached is controlling. When the quorum is reach,
+   the signing keys will be checked against the current signers list, and if a
+   collected signature's key is no longer on that list it is removed and
+   signatures will continue to be collected.
+   
+5) On the destination chain, the the owner of the "cross chain sequence number"
+   (see 1) can submit a "cross chain claim" transaction that includes the "cross
+   chain sequence number", the sidechain, and the destination. The "cross chain
+   sequence number" object must exist and must have already collected enough
+   signatures from the witness servers for this to succeed. On success, a
+   payment is made from the door account to the destination and the "cross chain
+   sequence number" is deleted. A "cross chain claim" transaction can only
+   succeed once, as the "cross chain sequence number" for that transaction can
+   only be created once. In case of error, the funds can be sent to an alternate
+   account and eventually returned to the initiating account. Note that this
+   transaction is only used if the optional destination account is not specified
+   in step (2) or there is an error when sending funds to that destination
+   account.
+
+6) When a witness server sees funds being claimed on the destination chain, it
+   submits a transaction on the source chain allowing the signature rewards to
+   be collected. It collects these signatures on the `XChainTransferRewardState`
+   created in step (2). The number of rewards will always equal the number of
+   signatures needed to reach a quorum. When a quorum of these signatures have
+   been collected, the rewards will be transferred to the destination addresses
+   specified in the signatures. (Note: the witness servers specify where the rewards
+   for its signature goes, this is not specified on the sidechain ledger
+   object). The `XChainTransferRewardState` is destroyed when the reward pool is
+   distributed.
+   
+The cross-chain transfer is now complete. Note that the transactions sent by the
+witness servers that add their signatures may send the signatures in a batch.
+
+## Supporting transactions overview
+
+In addition to the transactions used in a cross-chain transfer (described
+above), there are new transactions for creating a sidechain, changing sidechain
+parameters, and for using a cross-chain transfer to create a new account on the
+destination chain.
+
+The `SidechainCreate` transaction adds a sidechain ledger object to the account.
+This contains the two door accounts, the asset type that will be put into trust
+on the mainchain, the wrapped asset type on the sidechain, and the reward amount
+in XRP per signature. Optional, the minimum amount of XRP needed for an account
```

Comment:
do we supported non XRP assets such as tokens today?

Reply:
Yes

Reply:
that's great. I was under the assumption that only XRP assets were supported by the bridge in the new design